### PR TITLE
Do not clear input channels everytime focus changes

### PR DESCRIPTION
**Pull request by hultonha**
This change stops all inputs being cleared when focus changes between windows/widgets.

How things currently stand there's a problem when holding Alt (to begin camera orbit). If the user clicks outside the Viewport (e.g. changing selection of an entity in the outliner), then as the input is cleared it's as if Alt was released and clicking back in the viewport has the camera no longer orbit. This can be very disruptive to an artist/designer's flow, so I'd like to avoid it but at the same time I don't want this to cause any other issues.

I'd be very grateful to get people's feedback and ideas/suggestions and can potentially get QA involved too if they know of any existing edge cases or good things to test.

~I'm also going to investigate authoring a test for this which is why this is in draft status right now.~. I've now added some tests to codify this behavior.

Thanks!

Test output:

Before changes:

```
[ RUN      ] ViewportManipulatorControllerFixture.ChangingFocusDoesNotClearInput
D:/o3de/Code/Editor/Lib/Tests/test_ViewportManipulatorController.cpp(195): error: Value of: endedEvent
  Actual: true
Expected: false
[  FAILED  ] ViewportManipulatorControllerFixture.ChangingFocusDoesNotClearInput (97 ms)
[----------] 1 test from ViewportManipulatorControllerFixture (97 ms total)

[----------] Global test environment tear-down
[==========] 1 test from 1 test case ran. (99 ms total)
[  PASSED  ] 0 tests.
[  FAILED  ] 1 test, listed below:
[  FAILED  ] ViewportManipulatorControllerFixture.ChangingFocusDoesNotClearInput
```

After changes:

```
[==========] Running 3 tests from 1 test case.
[----------] Global test environment set-up.
[----------] 3 tests from ViewportManipulatorControllerFixture
[ RUN      ] ViewportManipulatorControllerFixture.AnEventIsNotPropagatedToTheViewportWhenAManipulatorHandlesItFirst
[       OK ] ViewportManipulatorControllerFixture.AnEventIsNotPropagatedToTheViewportWhenAManipulatorHandlesItFirst (2 ms)
[ RUN      ] ViewportManipulatorControllerFixture.ChangingFocusDoesNotClearInput
[       OK ] ViewportManipulatorControllerFixture.ChangingFocusDoesNotClearInput (107 ms)
[ RUN      ] ViewportManipulatorControllerFixture.ApplicationStateChangeDoesClearInput
[       OK ] ViewportManipulatorControllerFixture.ApplicationStateChangeDoesClearInput (38 ms)
[----------] 3 tests from ViewportManipulatorControllerFixture (150 ms total)

[----------] Global test environment tear-down
[==========] 3 tests from 1 test case ran. (152 ms total)
[  PASSED  ] 3 tests.
```

### CODE REVIEW

File: Code/Framework/AzToolsFramework/AzToolsFramework/Input/QtEventToAzInputManager.cpp [hunk 1]
```diff
@@ -255,6 +255,13 @@ namespace AzToolsFramework
             }
         }
 
+        // If the application state changes (e.g. we have alt-tabbed or minimized the
+        // main editor window) then ensure all input channels are cleared
+        if (eventType == QEvent::ApplicationStateChange)
```
Comment (pollend) [inline Code/Framework/AzToolsFramework/AzToolsFramework/Input/QtEventToAzInputManager.cpp:260]: do you want to filter to specific application states:

https://doc.qt.io/qt-5/qt.html#ApplicationState-enum

File: Code/Framework/AzToolsFramework/AzToolsFramework/Input/QtEventToAzInputManager.cpp [hunk 2]
```diff
@@ -246,7 +246,7 @@ namespace AzToolsFramework
 
         if (eventType == QEvent::Type::MouseMove)
```
Comment (pollend) [inline Code/Framework/AzToolsFramework/AzToolsFramework/Input/QtEventToAzInputManager.cpp:247]: could this be cleaned up by moving these cases into a switch block

File: Code/Framework/AzToolsFramework/AzToolsFramework/Input/QtEventToAzInputManager.cpp (hunk 1 above)
Comment (hultonha) [reply Code/Framework/AzToolsFramework/AzToolsFramework/Input/QtEventToAzInputManager.cpp:260]: I was doing something similar before (please see earlier comment from Nick and my reply) but decided clearing input for any kind of application state change makes sense (at least for now). On desktop most (all?) seem to be the 0x02 option.

File: Code/Framework/AzToolsFramework/AzToolsFramework/Input/QtEventToAzInputManager.cpp (hunk 2 above)
Comment (hultonha) [reply Code/Framework/AzToolsFramework/AzToolsFramework/Input/QtEventToAzInputManager.cpp:247]: I think it makes sense for that change to happen as a separate PR but it's not a bad idea for sure

**Comment by hultonha** [comment]
It seems like one of the tests I added is failing in AR but not locally 🤔 I'm not sure why... I'm going to investigate shortly

**Comment by hultonha** [comment]
AR has passed! 😄 Wound up using an explicit event to send the application state change from the test

```
QApplicationStateChangeEvent applicationStateChangeEvent(Qt::ApplicationState::ApplicationInactive);
QCoreApplication::sendEvent(m_rootWidget.get(), &applicationStateChangeEvent);
```

As opposed to using the `minimize` member function which did not appear to work as consistently - I'll merge first thing tomorrow.
